### PR TITLE
model/gemini: normalize tool schemas for function calling

### PR DESCRIPTION
**Pull request by bytethm**
close https://github.com/trpc-group/trpc-agent-go/issues/1252

## Summary by Sourcery

规范 Gemini 工具的函数调用模式（schema），以避免出现空的参数模式，并确保对象结构的一致性。

Bug 修复：
- 在发送给 Gemini 的请求中，对于没有输入模式的工具，避免发送空的 `parametersJsonSchema`。

增强：
- 在发送给 Gemini 之前，先对工具的输入和输出 JSON 模式进行规范化，确保对象模式始终包含 `properties` 字段。

测试：
- 新增测试，用于验证在序列化时，对于输入模式为 nil 的工具，会省略 `parametersJsonSchema` 字段。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Normalize Gemini tool schemas for function calling to avoid null parameter schemas and ensure consistent object structure.

Bug Fixes:
- Avoid sending a null parametersJsonSchema for tools without input schemas in Gemini requests.

Enhancements:
- Normalize tool input and output JSON schemas before sending them to Gemini, ensuring object schemas always include a properties field.

Tests:
- Add a test verifying that tools with nil input schemas omit the parametersJsonSchema field when marshaled.

</details>

### CODE REVIEW

File: model/gemini/gemini_test.go [hunk 1]
```diff
@@ -183,6 +183,30 @@ func (t *Tool) Declaration() *tool.Declaration {
...
 }
 
+func TestModel_convertTools_NilInputSchemaIsOmitted(t *testing.T) {
+	m := &Model{}
+	schema = nil
+
+	converted := m.convertTools(map[string]tool.Tool{
+		"tool": &Tool{},
+	})
+	if !assert.Len(t, converted, 1) {
```
Comment (Flash-LHR) [inline model/gemini/gemini_test.go]: 这里用 `if !assert.Len` 会静默 return，失败原因不够直观。建议改为 `require.Len` 并去掉分支。

Comment (bytethm) [reply model/gemini/gemini_test.go]: fixed

File: model/gemini/gemini.go [hunk 2]
```diff
@@ -522,12 +522,16 @@ func (m *Model) convertTools(tools map[string]tool.Tool) []*genai.Tool {
 	for _, t := range tools {
 		decl := t.Declaration()
 		funcDeclaration := &genai.FunctionDeclaration{
-			Description:          decl.Description,
-			Name:                 decl.Name,
-			ParametersJsonSchema: decl.InputSchema,
+			Description: decl.Description,
```
Comment (sandyskies) [inline model/gemini/gemini.go:525]: 再想想antropic是不是同样有问题？

Comment (bytethm) [reply model/gemini/gemini.go:525]: 看起来都有，不过每个 provider 的 schema 字段类型不同（any / string / map），没有统一的改法，需要各自适配一下。
